# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tsa/iptraf/asm_sys_types_h.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- iptraf-2.7.0/src/tr.c	2005-12-17 04:35:54.000000000 +0100
+++ src.iptraf.1134788240.15878.1654813972/iptraf-2.7.0/src/tr.c	2005-12-17 04:35:35.000000000 +0100
@@ -8,6 +8,16 @@
 
 #include <asm/types.h>
 #include <linux/if_tr.h>
+
+/* conflict between sys/types.h and asm/types.h */
+#define nlink_t libc_nlink_t
+#define mode_t libc_mode_t
+#define fd_set libc_fd_set
+#define dev_t libc_dev_t
+#define gid_t libc_gid_t
+#define uid_t libc_uid_t
+#define timer_t libc_timer_t
+
 #include <netinet/in.h>
 
 unsigned int get_tr_ip_offset(unsigned char *pkt)
